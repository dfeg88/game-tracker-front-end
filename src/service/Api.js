import axios from "axios";

export let api = () => axios.create({
  baseURL: "http://localhost:8080/",
  withCredentials: true
})