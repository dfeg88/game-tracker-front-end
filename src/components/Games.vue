<template>
  <v-layout row wrap class="ma-3" v-if="games">
      <v-flex xs12 sm6 md4 lg3
              v-for="(game, i) in games"
              :key="i">
        <Game :game="game"></Game>
      </v-flex>
  </v-layout>
</template>

<script>
import {api} from "@/service/Api";
import Game from "@/components/Game";

export default {
  name: "Games",
  components: {
    Game
  },
  data() {
    return {
      games: []
    }
  },
  async mounted() {
    this.games = (await api().get('v1/game')).data
    this.games.forEach(game => console.log(game))
  }
}
</script>

<style scoped>
.v-image__image{
  background-size:100% 100%;
}
</style>