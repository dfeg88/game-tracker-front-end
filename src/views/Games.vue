<template>
  <v-layout row wrap class="ma-3">
    <v-row v-if="games.length > 0">
      <v-flex v-for="(game, i) in games" :key="i" xs12 sm6 md4 lg3>
        <Game :id="game.name + '-' + game.platform" :game="game"></Game>
      </v-flex>
    </v-row>
    <v-flex v-else>
      <v-col xs12 sm6 md4 lg12 class="text-center">
        <p>It looks like you don't have any games..</p>
        <v-btn color="accent" elevation="8">
          <router-link :to="{name: 'create'}">Add Game</router-link>
        </v-btn>
      </v-col>
    </v-flex>
  </v-layout>
</template>

<script>
import {api} from "@/service/Api";
import Game from "@/components/Game";

export default {
  name: "Games",
  components: {
    Game
  },
  data() {
    return {
      games: []
    }
  },
  async mounted() {
    this.games = (await api().get('v1/game')).data
  }
}
</script>

<style scoped>
a, a:hover, a:active, a:visited {
  text-decoration: none;
  color: white;
}
</style>