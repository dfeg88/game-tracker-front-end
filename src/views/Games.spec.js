import {createLocalVue, mount} from "@vue/test-utils";
import Vuetify from "vuetify";
import Games from "@/views/Games";
import * as Api from '@/service/Api'

let wrapper;
const vuetify = new Vuetify();

describe('Games', () => {
  let get = jest.fn(() => Promise.resolve(
    {
      data: [
        {
          name: 'name',
          platform: 'platform',
          genre: 'genre',
          releaseDate: '01-01-2020',
          numberOfPlayers: 8,
          publisher: 'publisher',
        }
      ]
    }))

  Api.api = () => ({get})

  beforeEach(() => {
    wrapper = mount(Games, {
      localVue: createLocalVue(),
      vuetify,
      data() {
        return {
          games: []
        }
      }
    })
  });

  afterEach(() => {
    jest.clearAllMocks();
  })

  it('renders expected HTML', () => {
    expect(wrapper.elements).toMatchSnapshot();
  })

  it('displays games', async () => {
    expect(wrapper.vm.$data.games.length).toBe(1)
    expect(wrapper.find('#name-platform').exists).toBeTruthy();
  })

});