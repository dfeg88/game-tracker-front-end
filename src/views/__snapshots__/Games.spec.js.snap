// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Games renders expected HTML 1`] = `undefined`;
